src/main/java/org/jitsi/videobridge/Endpoint.java

```diff
@@ -638,9 +633,56 @@ public void addRtpExtension(RtpExtension rtpExtension)
     }
 
     @Override
-    public void setVideoConstraints(Map<String, VideoConstraints> newVideoConstraints)
+    public void setSenderVideoConstraints(ImmutableMap<String, VideoConstraints> newVideoConstraints)
     {
+        ImmutableMap<String, VideoConstraints>
+            oldVideoConstraints = bitrateController.getVideoConstraints();
+
         bitrateController.setVideoConstraints(newVideoConstraints);
+
+        Set<String> endpointsThatNoLongerCare
+            = new HashSet<>(oldVideoConstraints.keySet());
+        endpointsThatNoLongerCare.removeAll(newVideoConstraints.keySet());
+
+        // Endpoints that no longer care what they receive.
+        for (String id : endpointsThatNoLongerCare)
```

Comment:
I am probably misunderstanding something, but if these are endpoints that no longer have constraints, that means we may want to send them high quality layers, right? Why are we doing `removeReceiver`?

Reply:
> I am probably misunderstanding something, but if these are endpoints that no longer have constraints, that means we may want to send them high quality layers, right?

Currently endpoints that haven't signalled any video constraints will get 180p which is the default resolution that we've used since forever.

Reply:
Oh, I didn't know that.